### PR TITLE
Create failing test on bugfix that will be implemented

### LINKED ISSUE #1576: Maxlength Input Text & Textarea
When using safari's autocomplete feature on a inout field or textarea with maxchars enabled and the auto-filled content is longer than maxchars, the input field keeps the whole text.

When maxlength is reached, also via form autocomplete, the length should always be cut down to its max allowance

### PR DESCRIPTION
**Pull request by LucaMele**
Fixes #1576.

# Done is when (DoD):
- [x] I have added UI/Unit tests for my changes
- [x] Old tests and eslint rule are not broken
- [ ] I added modified component properties to the typescript typings
- [ ] Design has been reviewed with (here name of reviewer)
- [x] My changes are accordingly to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)
- [ ] My changes are well documented in the README and showcased in the stories 
- [x] My changes are shortly added to CHANGELOG.md with the issue number
- [ ] Tested with IE

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas (focus on **WHY**)
- [x] I have made/updated the ChangeLog Section in the README

### CODE REVIEW

File: src/components/10-atoms/input-text/react/story.jsx [hunk 1]
```diff
@@ -57,3 +55,60 @@ storyInputText.add('Input Text', () => {
...
+  const counter = text('counter', 'Still ##counter## characters left');
+  const counterMax = text('counterMax', 'Over character limit!');
+
+  const wrapper = document.createElement('div');
+  ReactDOM.render(
+    <form>
+      <AXAInputText
+        refid="fix-id-fake"
+        name="Name"
+      />
```
Comment (01231) [inline src/components/10-atoms/input-text/react/story.jsx:79]: for what is this input-field needed?

Comment (LucaMele) [reply src/components/10-atoms/input-text/react/story.jsx:79]: i will add a comment

Comment (LucaMele) [reply src/components/10-atoms/input-text/react/story.jsx:79]: done

**Review by 01231 — APPROVED** [review]
Nice!

File: src/components/10-atoms/input-text/index.js [hunk 2]
```diff
@@ -190,6 +189,22 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
...
+    // autocomplete instead is decreased all at once, therefore truncate it
+    if (this.charsLeft < 0 && this.maxLength) {
+      const { nativeInput } = this;
+      const value = nativeInput.value.substring(0, this.maxLength - 1);
+
+      // set value of native input element
+      nativeInput.value = value;
+      // set model value
+      this.modelValue = value;
+      // requestr update with the new value
```
Comment (danyball) [inline src/components/10-atoms/input-text/index.js:204]: typo requestr

File: src/components/10-atoms/input-text/index.js [hunk 3]
```diff
@@ -190,6 +189,22 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
     if (this.maxLength) {
       this.modelCounter = this.getCounterText;
     }
+
+    // means that a automatic prefelling function did fill it. Re-evaluate the component
+    // if a user types the chars, the charsLeft always decrease incrementally. With
+    // autocomplete instead is decreased all at once, therefore truncate it
+    if (this.charsLeft < 0 && this.maxLength) {
+      const { nativeInput } = this;
+      const value = nativeInput.value.substring(0, this.maxLength - 1);
```
Comment (danyball) [inline src/components/10-atoms/input-text/index.js:198]: You could rename variable to: valueCuttedToMaxLength for a better reading.

File: src/components/10-atoms/input-text/index.js [hunk 4]
```diff
@@ -190,6 +189,22 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
...
+
+      // set value of native input element
+      nativeInput.value = value;
+      // set model value
+      this.modelValue = value;
+      // requestr update with the new value
+      this.requestUpdate('value', value);
+      this.modelCounter = this.getCounterText;
```
Comment (danyball) [inline src/components/10-atoms/input-text/index.js:206]: Could you move code from line 189-191 to line 209? If you could, this line 206 is not necessary.

File: src/components/10-atoms/input-text/ui.test.js [hunk 5]
```diff
@@ -146,3 +146,18 @@ test('should not show counter text if counter (text) not set', async t => {
...
+  `${host}/iframe.html?id=atoms-input-text-react--input-text-simulate-autocomplete`
+);
+
+test('should cut text when autocomplete sets value over maxLength', async t => {
+  // in the story, the autocomplete function is only simulated after a timeout, therefore wait here
+  await t.wait( 2000 );
+  const inputValue = await ClientFunction(() =>
+    document.querySelector('#fix-id-86452623').value
+  );
+  await t.wait( 1000 );
```
Comment (danyball) [inline src/components/10-atoms/input-text/ui.test.js:160]: These hard waits costs us 3 seconds in our build. I would try to remove them. A first idea for this:

Move the "setTimeout" Code from story.jsx to this test and use `await`

**Pull request merged**